### ISSUE TITLE
Chatterbot install error.

### ISSUE THREAD

**Issue by LS255** [numpy/numpy#16112]
```
C:\WINDOWS\system32>pip3 install chatterbot
Collecting chatterbot
  Using cached ChatterBot-1.0.5-py2.py3-none-any.whl (67 kB)
Collecting python-dateutil<2.8,>=2.7
  Using cached python_dateutil-2.7.5-py2.py3-none-any.whl (225 kB)
Collecting pymongo<4.0,>=3.3
  Using cached pymongo-3.10.1-cp38-cp38-win_amd64.whl (355 kB)
Collecting pyyaml<5.2,>=5.1
  Using cached PyYAML-5.1.2.tar.gz (265 kB)
Collecting pytz
  Using cached pytz-2020.1-py2.py3-none-any.whl (510 kB)
Collecting mathparse<0.2,>=0.1
  Using cached mathparse-0.1.2-py3-none-any.whl (7.2 kB)
Collecting spacy<2.2,>=2.1
  Using cached spacy-2.1.9.tar.gz (30.7 MB)
  Installing build dependencies ... error
  ERROR: Command errored out with exit status 1:
   command: 'c:\users\l. s. (alas)\appdata\local\programs\python\python38\python.exe' 'c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\pip' install --ignore-installed --no-user --prefix 'C:\Users\L. S. (Alas)\AppData\Local\Temp\pip-build-env-1azxmhzx\overlay' --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- setuptools 'wheel>0.32.0,<0.33.0' Cython 'cymem>=2.0.2,<2.1.0' 'preshed>=2.0.1,<2.1.0' 'murmurhash>=0.28.0,<1.1.0' 'thinc>=7.0.8,<7.1.0'
       cwd: None
  Complete output (117 lines):
  Collecting setuptools
    Using cached setuptools-46.1.3-py3-none-any.whl (582 kB)
  Collecting wheel<0.33.0,>0.32.0
    Using cached wheel-0.32.3-py2.py3-none-any.whl (21 kB)
  Collecting Cython
    Using cached Cython-0.29.17-cp38-cp38-win_amd64.whl (1.7 MB)
  Collecting cymem<2.1.0,>=2.0.2
    Using cached cymem-2.0.3-cp38-cp38-win_amd64.whl (33 kB)
  Collecting preshed<2.1.0,>=2.0.1
    Using cached preshed-2.0.1.tar.gz (113 kB)
  Collecting murmurhash<1.1.0,>=0.28.0
    Using cached murmurhash-1.0.2-cp38-cp38-win_amd64.whl (20 kB)
  Collecting thinc<7.1.0,>=7.0.8
    Using cached thinc-7.0.8.tar.gz (1.9 MB)
      ERROR: Command errored out with exit status 1:
       command: 'c:\users\l. s. (alas)\appdata\local\programs\python\python38\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\L. S. (Alas)\\AppData\\Local\\Temp\\pip-install-9bjuv19t\\thinc\\setup.py'"'"'; __file__='"'"'C:\\Users\\L. S. (Alas)\\AppData\\Local\\Temp\\pip-install-9bjuv19t\\thinc\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base 'C:\Users\L. S. (Alas)\AppData\Local\Temp\pip-pip-egg-info-5wegqp2a'
           cwd: C:\Users\L. S. (Alas)\AppData\Local\Temp\pip-install-9bjuv19t\thinc\
      Complete output (97 lines):
      Processing numpy/random\_bounded_integers.pxd.in
      Processing numpy/random\mtrand.pyx
      Traceback (most recent call last):
        File "C:\Users\LS36F4~1.(AL\AppData\Local\Temp\easy_install-9gzaz9v9\numpy-1.18.3\tools\cythonize.py", line 61, in process_pyx
          from Cython.Compiler.Version import version as cython_version
      ModuleNotFoundError: No module named 'Cython'

      During handling of the above exception, another exception occurred:

      Traceback (most recent call last):
        File "C:\Users\LS36F4~1.(AL\AppData\Local\Temp\easy_install-9gzaz9v9\numpy-1.18.3\tools\cythonize.py", line 238, in <module>
          main()
        File "C:\Users\LS36F4~1.(AL\AppData\Local\Temp\easy_install-9gzaz9v9\numpy-1.18.3\tools\cythonize.py", line 234, in main
          find_process_files(root_dir)
        File "C:\Users\LS36F4~1.(AL\AppData\Local\Temp\easy_install-9gzaz9v9\numpy-1.18.3\tools\cythonize.py", line 225, in find_process_files
          process(root_dir, fromfile, tofile, function, hash_db)
        File "C:\Users\LS36F4~1.(AL\AppData\Local\Temp\easy_install-9gzaz9v9\numpy-1.18.3\tools\cythonize.py", line 191, in process
          processor_function(fromfile, tofile)
        File "C:\Users\LS36F4~1.(AL\AppData\Local\Temp\easy_install-9gzaz9v9\numpy-1.18.3\tools\cythonize.py", line 66, in process_pyx
          raise OSError('Cython needs to be installed in Python as a module')
      OSError: Cython needs to be installed in Python as a module
      Running from numpy source directory.
      C:\Users\LS36F4~1.(AL\AppData\Local\Temp\easy_install-9gzaz9v9\numpy-1.18.3\setup.py:461: UserWarning: Unrecognized setuptools command, proceeding with generating Cython sources and expanding templates
        run_build = parse_setuppy_commands()
      Traceback (most recent call last):
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\sandbox.py", line 154, in save_modules
          yield saved
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\sandbox.py", line 195, in setup_context
          yield
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\sandbox.py", line 250, in run_setup
          _execfile(setup_script, ns)
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\sandbox.py", line 45, in _execfile
          exec(code, globals, locals)
        File "C:\Users\LS36F4~1.(AL\AppData\Local\Temp\easy_install-9gzaz9v9\numpy-1.18.3\setup.py", line 488, in <module>
        File "C:\Users\LS36F4~1.(AL\AppData\Local\Temp\easy_install-9gzaz9v9\numpy-1.18.3\setup.py", line 469, in setup_package
        File "C:\Users\LS36F4~1.(AL\AppData\Local\Temp\easy_install-9gzaz9v9\numpy-1.18.3\setup.py", line 275, in generate_cython
      RuntimeError: Running cythonize failed!

      During handling of the above exception, another exception occurred:

      Traceback (most recent call last):
        File "<string>", line 1, in <module>
        File "C:\Users\L. S. (Alas)\AppData\Local\Temp\pip-install-9bjuv19t\thinc\setup.py", line 261, in <module>
          setup_package()
        File "C:\Users\L. S. (Alas)\AppData\Local\Temp\pip-install-9bjuv19t\thinc\setup.py", line 201, in setup_package
          setup(
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\__init__.py", line 144, in setup
          _install_setup_requires(attrs)
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\__init__.py", line 139, in _install_setup_requires
          dist.fetch_build_eggs(dist.setup_requires)
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\dist.py", line 716, in fetch_build_eggs
          resolved_dists = pkg_resources.working_set.resolve(
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\pkg_resources\__init__.py", line 780, in resolve
          dist = best[req.key] = env.best_match(
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\pkg_resources\__init__.py", line 1065, in best_match
          return self.obtain(req, installer)
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\pkg_resources\__init__.py", line 1077, in obtain
          return installer(requirement)
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\dist.py", line 786, in fetch_build_egg
          return cmd.easy_install(req)
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\command\easy_install.py", line 679, in easy_install
          return self.install_item(spec, dist.location, tmpdir, deps)
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\command\easy_install.py", line 705, in install_item
          dists = self.install_eggs(spec, download, tmpdir)
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\command\easy_install.py", line 890, in install_eggs
          return self.build_and_install(setup_script, setup_base)
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\command\easy_install.py", line 1158, in build_and_install
          self.run_setup(setup_script, setup_base, args)
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\command\easy_install.py", line 1144, in run_setup
          run_setup(setup_script, args)
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\sandbox.py", line 253, in run_setup
          raise
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\contextlib.py", line 131, in __exit__
          self.gen.throw(type, value, traceback)
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\sandbox.py", line 195, in setup_context
          yield
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\contextlib.py", line 131, in __exit__
          self.gen.throw(type, value, traceback)
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\sandbox.py", line 166, in save_modules
          saved_exc.resume()
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\sandbox.py", line 141, in resume
          six.reraise(type, exc, self._tb)
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\_vendor\six.py", line 685, in reraise
          raise value.with_traceback(tb)
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\sandbox.py", line 154, in save_modules
          yield saved
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\sandbox.py", line 195, in setup_context
          yield
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\sandbox.py", line 250, in run_setup
          _execfile(setup_script, ns)
        File "c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\setuptools\sandbox.py", line 45, in _execfile
          exec(code, globals, locals)
        File "C:\Users\LS36F4~1.(AL\AppData\Local\Temp\easy_install-9gzaz9v9\numpy-1.18.3\setup.py", line 488, in <module>
        File "C:\Users\LS36F4~1.(AL\AppData\Local\Temp\easy_install-9gzaz9v9\numpy-1.18.3\setup.py", line 469, in setup_package
        File "C:\Users\LS36F4~1.(AL\AppData\Local\Temp\easy_install-9gzaz9v9\numpy-1.18.3\setup.py", line 275, in generate_cython
      RuntimeError: Running cythonize failed!
      Cythonizing sources
      ----------------------------------------
  ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  ----------------------------------------
ERROR: Command errored out with exit status 1: 'c:\users\l. s. (alas)\appdata\local\programs\python\python38\python.exe' 'c:\users\l. s. (alas)\appdata\local\programs\python\python38\lib\site-packages\pip' install --ignore-installed --no-user --prefix 'C:\Users\L. S. (Alas)\AppData\Local\Temp\pip-build-env-1azxmhzx\overlay' --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- setuptools 'wheel>0.32.0,<0.33.0' Cython 'cymem>=2.0.2,<2.1.0' 'preshed>=2.0.1,<2.1.0' 'murmurhash>=0.28.0,<1.1.0' 'thinc>=7.0.8,<7.1.0' Check the logs for full comm
```

I have c++ vs build tools.

**Comment by anirudh2290** [comment]
```
ModuleNotFoundError: No module named 'Cython'
```
not sure what the exact issue is and if this is even related to numpy. Having said that have you tried `python3 -m pip install cython` ?

**Comment by LS255** [comment]
It's already done.

**Comment by eric-wieser** [comment]
It looks like maybe something goes wrong if a package has a build dependency on numpy, and cython is not yet installed.

Does installing cython first fix the problem? What does `pip -V` give?

**Comment by LS255** [comment]
Can you give me complete information Starting to End. Install package to
requirements all thing please.
Please😫🙏🙏💓🙏🙏💓.
    It's will be most useful for me please. Please please please please
please please help.help.help.help.help. please.........

**Comment by LS255** [comment]
Sorry, I forget to give you a great Thanks🙏🙇🙏💕🙏🙇 bro...
    Thank you so much for your useful help.
 Really really Thanks.....

**Comment by anirudh2290** [comment]
@LS255 I tried your steps in a venv which didnt have cython installed and I was able to reproduce the issue. After installing cython the issue was fixed. I ran the following two commands:

```
python3 -m pip install cython
python3 -m pip install chatterbot
```
The second step takes quite a while because of some of the build dependencies.

If the above steps don't work for you can you please let us know the pip version with `pip -V` as asked by @eric-wieser .

**Comment by LS255** [comment]
Thank you so much for your support.
And you help.
  But my problem waa solved by installing python 3.7.4   actually
chatterbot in not Confetable with python 3.8.2y problem aolved after
replacing this.
